js/src/dropdown.js

```diff
@@ -405,7 +405,7 @@ class Dropdown extends BaseComponent {
 
     event.preventDefault()
 
-    const getToggleButton = SelectorEngine.findOne(SELECTOR_DATA_TOGGLE, event.delegateTarget.parentNode)
+    const getToggleButton = this.matches(SELECTOR_DATA_TOGGLE) ? this : SelectorEngine.prev(this, SELECTOR_DATA_TOGGLE)[0]
```

Comment:
As on Line 98 you search prev & next, I suppose we should keep the same attitude here.

In general, I would prefer to keep the right markup solution fixing the docs and adding it in migration notes, instead of doing tricks on javascipt.
I don't know if we would count that, as a breaking change but the specific example/problem, negates the suggested markup 

@cc @twbs/team